adal/src/main/java/com/microsoft/aad/adal/IBrokerProxy.java

```diff
@@ -37,12 +37,16 @@
      * @return True package is available and authenticator is installed at
      *         Account manager
      */
-    boolean canSwitchToBroker();
+    BrokerProxy.SwitchToBroker canSwitchToBroker();
     
     boolean verifyUser(String username, String uniqueid);
 
     boolean canUseLocalCache();
 
+    boolean verifyBrokerPermissionsAPI22AndLess() throws UsageAuthenticationException;
```

Comment:
They cannot be removed if we keep the declaration of mBrokerProxy in AuthenticationContext. Because verifyBrokerPermissionsAPI22AndLess() is called in AuthenticationContext, and the property mBrokerProxy is defined as below:
*private IBrokerProxy mBrokerProxy = null;*
We could not call these two functions if they are not defined in the interface IBrokerProxy.

One possible solution is to change to:
*private BrokerProxy mBrokerProxy = null;*